File: ObjectAvoidance/java/src/main/java/elite/math/trig/Arccosine.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Hypotenuse;

/**
 *
 * @author CFerg (Elite)
 */
public class Arccosine {

    /**
     *
     */
    public double arccosine;
    
    /**
     *
     * @param h
     * @param a
     */
    public Arccosine(Hypotenuse h, Adjacent a){
        this.arccosine = (h.hypotenuse / a.adjacent);
    }
    
    /**
     *
     * @param cos
     */
    public Arccosine(Cosine cos){
        this.arccosine = (1 / cos.cosine);
    }
    
    /**
     *
     * @param sec
     */
    public Arccosine(Secant sec){
        this.arccosine = sec.secant;
    }
}

File: ObjectAvoidance/kotlin/src/elite/math/trig/Arccosine.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Hypotenuse

/**
 *
 * @author CFerg (Elite)
 */
class Arccosine {

    /**
     *
     */
    var arccosine: Double = 0.toDouble()

    /**
     *
     * @param h
     * @param a
     */
    constructor(h: Hypotenuse, a: Adjacent) {
        this.arccosine = h.hypotenuse / a.adjacent
    }

    /**
     *
     * @param cos
     */
    constructor(cos: Cosine) {
        this.arccosine = 1 / cos.cosine
    }

    /**
     *
     * @param sec
     */
    constructor(sec: Secant) {
        this.arccosine = sec.secant
    }
}